src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlExceptionActionProcessor.cs

```diff
@@ -33,6 +36,10 @@ public Task Execute(TRequest request, TException exception, CancellationToken ca
                 {
                     throw new SqlQueryPlanException(Core.Resources.SqlQueryProcessorRanOutOfInternalResourcesException);
                 }
+                else if (sqlException.Number == DeadlockVictim)
```

Comment:
I thought the plan was also to not expose these errors to the customer. I didn't happen to see any changes related to that here, unless I missed something.

Reply:
What message do we want to send to the customer?
Also, what do you think the HTTP status code should be that gets returned?

Reply:
Since its sql contention perhaps 409?

Reply:
updated to a 409

Reply:
> I thought the plan was also to not expose these errors to the customer. I didn't happen to see any changes related to that here, unless I missed something.

I have revised the message for the deadlock in the case that retries don't work for whatever reason. For those, the message can be viewed in the Resources file that is also part of this PR.